### PR TITLE
several MIP improvements

### PR DESCRIPTION
**Pull request by lgottwald**
Contains several MIP improvements, some code cleanup, and several bugfixes.

### CODE REVIEW

**Comment by jajhall** [comment]
Lots of CI failures. Related to recent changes to CI @galabovaa? 

Could @lgottwald and @galabovaa coordinate to identify the cause and fix?

File: src/util/HighsSparseVectorSum.h [hunk 1]
```diff
@@ -31,50 +30,43 @@ class HighsSparseVectorSum {
...
+    if (values[index] != 0.0) {
       values[index] += value;
     } else {
       values[index] = value;
-      nonzeroflag[index] = true;
       nonzeroinds.push_back(index);
     }
+
+    if (values[index] == 0.0)
+      values[index] = std::numeric_limits<double>::min();
```
Comment (mmuetzel) [inline src/util/HighsSparseVectorSum.h:47]: Missing `#include <limits>`. See: https://en.cppreference.com/w/cpp/types/numeric_limits

**Comment by lgottwald** [comment]
What is the fail with the python interface? I did not touch that

**Comment by jajhall** [comment]
It's because the changes made by Stefan haven't been propagated to the Python bindings

**Review by jajhall — APPROVED** [review]
I'm OK with the changes

**Comment by jajhall** [comment]
We'd better fix the Python bindings fairly soon now that master fails the CI tests!

**Pull request merged**